pkg/engine/exceptions.go

```diff
@@ -53,6 +55,23 @@ func matchesException(
 		resource = policyContext.OldResource()
 	}
 	for _, candidate := range candidates {
+		if candidate.Spec.HasPodSecuruty() {
+			if rule.HasValidatePodSecurity() {
+				for _, pss := range rule.Validation.PodSecurity.Exclude {
+					if wildcard.Match(pss.ControlName, candidate.Spec.PodSecurity.ControlName) {
+						for _, iref := range pss.Images {
+							for _, i := range candidate.Spec.PodSecurity.Images {
+								if wildcard.Match(iref, i) {
+									return candidate, nil
+								}
+							}
+						}
+					}
+				}
```

Comment:
I think this isn't the correct approach. We should re-use the [logic](https://github.com/kyverno/kyverno/blob/main/pkg/pss/evaluate.go#L64) where it exempt specific controls within a profile.

The user will either use `validate.podSecurity.exclude` or write a policy exception to do the same exact thing.

Reply:
Makes sense, so the logic should look something like this?
Check if the `validate.podSecurity.exclude.images` and `spec.podSecurity.images` have the same images, if yes, then we can just simply return, but if the images do not match then simply apply the `exemptKyvernoExclusion()` over it, am I right?

Reply:
I believe the user can't use both `validate.podSecurity.exclude.images` and `spec.podSecurity.images` and if that's the case, then this is nothing to do with PolicyExceptions because it should always apply `exemptKyvernoExclusion()` if `spec.podSecurity` is defined.

Reply:
To be more clear, you should use [EvaluatePod()](https://github.com/kyverno/kyverno/blob/main/pkg/pss/evaluate.go#L108) because it applies both pod level checks and container level checks.

In that case, `spec.podSecurity.images` will be optional in case of a control applied to a pod level, otherwise it is required.

Reply:
To be more clear, you should use [EvaluatePod()](https://github.com/kyverno/kyverno/blob/main/pkg/pss/evaluate.go#L108) because it applies both pod level checks and container level checks.

In that case, `spec.podSecurity.images` will be optional in case of a control applied to a pod level, otherwise it is required.

Reply:
You need this part only in PolicyExceptions:
https://github.com/kyverno/kyverno/blob/51fc790aa77376681dc506d441b134093f6d0687/pkg/pss/evaluate.go#L116-L130

Reply:
I am getting a little bit confused here, starting from the `matchesException()` function in `pkg/engine/exceptions.go`, we want to return an exception that is being applied to a particular resource right?
Now in this function, we'll check if user has defined `spec.podSecurity` or not, after that, we'll pass the `podSecurity` rule to the above mentioned logic, which will give us `defaultCheckResults`, but what after that? Should we check if `len(defaultCheckResults) == 0` and if it is, then return the `candidate` from the `matchesException()` function or how?

Reply:
> You need this part only in PolicyExceptions:
> 
> https://github.com/kyverno/kyverno/blob/51fc790aa77376681dc506d441b134093f6d0687/pkg/pss/evaluate.go#L116-L130

If we are going with only this part of the code, what would be the value of `pod` and `levelVersion`? For `exclude` we can use `candidate.Spec.PodSecurity` ig.

Reply:
One solution is to use the `EvaluatePod()` in the `matchesException()`. It isn't the best approach though because you have to figure out how to indicate if there's a policy exception or not.

Starting from the `matchesException()`, it mainly finds all exceptions i.e. candidates for a specific policy and rule and then it loops through all candidates. 
For each candidate, it checks the `match` block i.e. checks if the resource matches the `policyexception.spec.match`, if there is no error then it is a match and we return the policy exception.
Instead of returning the policy exception, we will check if the rule is `validate.podSecurity` and the policy exception has `spec.podSecurity`, then we will call `EvaluatePod()`. Otherwise, we will return the policy exception as it is done today.
```
// if there's no error it means a match
if err == nil {
   if rule.HasValidatePodSecurity() && candidate.Spec.PodSecurity {
         // copy the rule
	 r := rule.DeepCopy()

         // set the validation.podsecurity.exclude in the rule
         r.Validation.PodSecurity.Exclude = candidate.Spec.PodSecurity

         // create an instance of the pod from the resource.
         // call EvaluatePod()	
         // if EvaluatePod() return true, then we have an exception.	 
    }
    return candidate, nil
}
```

I amn't sure if this will work though. So it is better to understand how we apply pod security checks at pkg/pss/evaluate.go and then you can try to think about the best approach to apply policy exceptions.

Reply:
In addition, if we have no policy exceptions, then we will apply the pod security checks again. We will end up applying this logic twice. 
The main issue is that we need `defaultCheckResults` so that we can remove from it the excluded controls and to get the `defaultCheckResults` we need to actually apply the PSS checks.

@realshuting - Do you have any idea how to apply pod security exemptions through policy exceptions?
Given that the policy exception will have the following schema: https://github.com/kyverno/kyverno/blob/27e9dc541a6980f9c8b6d522745618a51fc00fc5/api/kyverno/v2alpha1/policy_exception_types.go#L63-L78